### PR TITLE
SWARM-1548: S2I Build meta data

### PR DESCRIPTION
**Pull request by heiko-braun**

### CODE REVIEW

File: .openshiftio/application.yaml [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    - type: ConfigChange
+    - imageChange: {}
+      type: ImageChange
+  status:
+    lastVersion: 0
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      fabric8.io/scm-con-url: scm:git:https://github.com/openshiftio/booster-parent.git/wfswarm-rest-http
```
Comment (kenfinnigan) [inline .openshiftio/application.yaml]: The scm urls seem a bit odd here

File: .openshiftio/application.yaml [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      prometheus.io/port: "9779"
+      fabric8.io/scm-url: https://github.com/openshiftio/wfswarm-rest-http
+      fabric8.io/iconUrl: img/icons/wildfly.svg
+      fabric8.io/git-branch: SWARM-1548
```
Comment (kenfinnigan) [inline .openshiftio/application.yaml]: Does the command need to be run on the master branch and then a new branch created, to not have the branch here?

File: .openshiftio/application.yaml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      prometheus.io/scrape: "true"
+      fabric8.io/scm-devcon-url: scm:git:git:@github.com:openshiftio/booster-parent.git/wfswarm-rest-http
+      fabric8.io/scm-tag: booster-parent-10
```
Comment (kenfinnigan) [inline .openshiftio/application.yaml]: Is it meant to grab the tag of the parent?

File: .openshiftio/application.yaml [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    labels:
+      expose: "true"
+      app: wfswarm-rest-http
+      provider: fabric8
+      version: 8-SNAPSHOT
```
Comment (kenfinnigan) [inline .openshiftio/application.yaml]: Does the `version` pose a problem for our pom-redhat.xml?

**Comment by heiko-braun** [comment]
@kenfinnigan This is an improved version that has a lot of the fabric8 cruft removed (annotations, labels, etc).

**Comment by heiko-braun** [comment]
With respect to maintenance this should be better, but still there is a lot boilerplate necessary to make s2i work. Long term we need to think how to automate this.

**Comment by heiko-braun** [comment]
Sorry, accidentally closed this one

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**